pkg/installer/custominstallconfig.go

```diff
@@ -63,6 +69,8 @@ func (m *manager) applyInstallConfigCustomisations(installConfig *installconfig.
 	if m.oc.Properties.NetworkProfile.GatewayPrivateEndpointIP != "" {
 		dnsConfig.GatewayPrivateEndpointIP = m.oc.Properties.NetworkProfile.GatewayPrivateEndpointIP
 		dnsConfig.GatewayDomains = m.getGatewayDomains(m.env, m.oc)
+		localdnsConfig.GatewayPrivateEndpointIP = m.oc.Properties.NetworkProfile.GatewayPrivateEndpointIP
+		localdnsConfig.GatewayDomains = m.getGatewayDomains(m.env, m.oc)
 	}
```

Comment:
Capturing slack conversation for future selves if needed: The duplication between localdnsConfig and dnsConfig is inteded as a temporary artifact to keep things working as they are until after other PRs land, then we will remove the duplication.